### PR TITLE
config: add sbom flags documentation in sample config file

### PR DESCRIPTION
**Pull request by jinroh**
### What does this PR do?

Adds description and some documentation in our sample config file for `sbom.*` options.

### Motivation

These fields are missing.

### CODE REVIEW

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -1470,6 +1470,42 @@ api_key:
...
+
+  ## @param enabled - boolean - optional - default: false
+  ## @env DD_SBOM_ENABLED - boolean - optional - default: false
+  ## Set to true to enable SBOM generation.
+  #
+  # enabled: false
+
+  ## @param container_image - custom object - optional
+  ## Configuration for generating SBOM for container images.
+  #
```
Comment (0intro) [inline pkg/config/config_template.yaml:1491]: Add `container_image` section.

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -1470,6 +1470,42 @@ api_key:
...
+    ## @param enabled - boolean - optional - default: false
+    ## @env DD_SBOM_CONTAINER_IMAGE_ENABLED - boolean - optional - default: false
+    ## Set to true to enable SBOM generation for container images.
+    #
+    # enabled: false
+
+
```
Comment (0intro) [inline pkg/config/config_template.yaml:1498]: Remove empty line.

File: pkg/config/config_template.yaml [hunk 3]
```diff
@@ -1470,6 +1470,42 @@ api_key:
...
+  ## @param host - custom object - optional
+  ## Configuration for generating SBOM for the host.
+  #
```
Comment (0intro) [inline pkg/config/config_template.yaml:1501]: Add `host` section.

**Review by 0intro — APPROVED** [review]

File: pkg/config/config_template.yaml
Comment (L3n41c) [inline pkg/config/config_template.yaml:1]: It seems there are other SBOM related parameters honored by the agent that are not documented in the template:
https://github.com/DataDog/datadog-agent/blob/3fa4f01592c5444c303cd2a11bc4106d6f5a995e/pkg/config/setup/config.go#L813-L836

**Review by jeremy-hanna — APPROVED** [review]
👍

File: pkg/config/config_template.yaml [hunk 4]
```diff
@@ -1470,6 +1470,43 @@ api_key:
...
   {{ end -}}
 {{ end -}}
+
+{{- if .SBOM }}
+#####################
+## SBOM Generation ##
+#####################
+
+## @param sbom - custom object - optional
+## Enter specific configurations for generating Software Bill of Materials (SBOM).
```
Comment (buraizu) [inline pkg/config/config_template.yaml:1480]:
```suggestion
## Enter specific configurations for generating a Software Bill of Materials (SBOM).
```

File: pkg/config/config_template.yaml [hunk 5]
```diff
@@ -1470,6 +1470,43 @@ api_key:
...
+  #
+  # container_image:
+    ## @param enabled - boolean - optional - default: false
+    ## @env DD_SBOM_CONTAINER_IMAGE_ENABLED - boolean - optional - default: false
+    ## Set to true to enable SBOM generation for container images.
+    #
+    # enabled: false
+
+  ## @param host - custom object - optional
+  ## Configuration for generating SBOM for the host.
```
Comment (buraizu) [inline pkg/config/config_template.yaml:1500]:
```suggestion
  ## Configuration for generating an SBOM for the host.
```

**Review by buraizu — APPROVED** [review]
Approving with two minor suggestions

File: pkg/config/config_template.yaml
Comment (jinroh) [reply pkg/config/config_template.yaml:1]: I intentionally left out the more advanced configuration parameters on purpose for now.

I'm not sure how to describe what they do precisely, and if we should expose them at all. Happy to have a more detailed description from your team.

**Pull request closed without merging**